src/cryfs-cli/program_options/ProgramOptions.cpp

```diff
@@ -16,6 +16,20 @@ ProgramOptions::ProgramOptions(const bf::path &baseDir, const bf::path &mountDir
     :_baseDir(baseDir), _mountDir(mountDir), _configFile(configFile), _foreground(foreground),
      _cipher(cipher), _blocksizeBytes(blocksizeBytes), _unmountAfterIdleMinutes(unmountAfterIdleMinutes),
      _logFile(logFile), _fuseOptions(fuseOptions) {
+
+    auto hasNoOption = [&](const char *opt) {
+        for (const string& it : _fuseOptions) {
+            if (std::strncmp(it.c_str(), opt, std::strlen(opt))) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (hasNoOption("subtype=cryfs") && hasNoOption("fsname=cryfs@")) {
+               _fuseOptions.push_back("-ofsname=cryfs@"+baseDir.native());
+               _fuseOptions.push_back("-osubtype=cryfs");
+    }
```

Comment:
```Another option might be to pass in two string arguments to the Fuse constructor, one for the file system type (i.e. "-osubtype") and one for the file system source (i.e. "-ofsname")```

My latest commit went with this option.

I have "fsname" as an optional argument because the Fuse class does not require it.